staging/src/k8s.io/apiserver/pkg/server/routes/profiling.go

```diff
@@ -41,3 +44,39 @@ func redirectTo(to string) func(http.ResponseWriter, *http.Request) {
 		http.Redirect(rw, req, to, http.StatusFound)
 	}
 }
+
+type SocketProfiling struct {
+	path string
+	mux  *http.ServeMux
+}
+
+func NewSocketProfiling(path string) *SocketProfiling {
+	return &SocketProfiling{
+		path: path,
+		mux:  http.NewServeMux(),
+	}
+}
+
+func (s *SocketProfiling) Install() {
+	s.mux.HandleFunc("/debug/pprof", redirectTo("/debug/pprof/"))
+	s.mux.HandleFunc("/debug/pprof/", pprof.Index)
+	s.mux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	s.mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	s.mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	s.mux.HandleFunc("/debug/pprof/trace", pprof.Trace)
+}
+
+func (s *SocketProfiling) Run() error {
+	if err := os.RemoveAll(s.path); err != nil {
```

Comment:
Is this dangerous removing all directories and subdirectories with the corresponding path? Maybe just `os.Remove` would be more suitable?